core/src/main/java/org/opensearch/sql/ast/expression/FunctionWithCachedValue.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.ast.expression;
+
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import org.opensearch.sql.ast.AbstractNodeVisitor;
+
+/**
+ * Expression node that holds a function which should be replaced by its constant[1] value.
+ * [1] Constant at execution time.
+ */
+@EqualsAndHashCode(callSuper = false)
+public class FunctionWithCachedValue extends Function {
```

Comment:
These are called _constant functions_ in math. 
`ConstantFunction` or `ConstantValueFunction` better explain what's special in this case.

Reply:
Thank you, renamed in 6ef778de.